dev/WindowsAppSDK_Insights/WindowsAppSDKInsights.h

```diff
@@ -10,15 +10,64 @@
 #error "WIL Tracelogging.h must not be explicitly included when including this file"
 #endif
 
-char* WINDOWSAPPSDK_PACKAGE_VER();
-char* WINDOWSAPPSDK_EXPERIMENTATION_LEVEL();
+#include <wil/resource.h>
+#include <string>
+
+namespace Microsoft
```

Comment:
NIT: C++ allows you to specify multiple namespaces in 1 statement e.g. `namespace A::B::C`

As of C++17 I think

Reply:
Do we recommend C++ 17 for our consumers?

Reply:
Is Insights a public API or just for consumption within WinAppSDK?

I think we (everyone in WinAppSDK) are (or can) use C++17. Worst case we can tweak Insights' .h if someone finds a real blocker.

Publicly I'm not sure what level of C++ we're requiring, but definitely something. More than C is required e.g. MsixDynamicDependency.h declares `enum class foo : uint32_t...` - that `class` keyword requires C++, and that's intentionally used per API Review feedback/requirements.

If WindowsAppSDKInsights.h is a public API there should be a spec under `specs\insights` that went through API Review. Not such beast so I assumed it's an internal API, and in the Transport package so DWriteCore etc up the stack w/in WinAppSDK can call it (not because it's meant for public consumption). But perhaps I'm mistaken

Reply:
It is technically public since we are publishing it with WindowsAppSDK. However, it is most useful for WindowsAppSDK code since it adds most value here.